frontend/src/OnBoardingForm/ContinueButton.jsx

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import EnterIcon from '../../assets/images/onboardingassets/Icons/Enter';
 import Spinner from '@/_ui/Spinner';
 
-function ContinueButton({
-  buttonState,
-  setPage,
-  setButtonState,
-  formData,
-  page,
-  setCompleted,
-  isLoading,
-  setIsLoading,
-  darkMode,
-}) {
+function ContinueButton({ setPage, formData, page, setCompleted, isLoading, setIsLoading, darkMode }) {
+  const activeCondition =
+    isLoading ||
+    (page == 0 && !formData.companyName) ||
+    (page == 0 && formData.companyName.trim().length === 0) ||
+    (page == 1 && !formData.role) ||
+    (page == 2 && !formData.companySize);
+
+  useEffect(() => {
+    const keyDownHandler = (event) => {
+      if (event.key === 'Enter' && !activeCondition) {
+        setPage((currPage) => currPage + 1);
```

Comment:
What if the user presses the key when he is on page 2? IMO. its better to do it like this 
         `page != 2 && setPage((currPage) => currPage + 1);`